gammapy/modeling/scipy.py

```diff
@@ -164,12 +162,26 @@ def stat_profile_ul_scipy(
     """
     interp = interpolate_profile(value_scan, stat_scan, interp_scale=interp_scale)
 
+    if np.allclose(stat_scan, stat_scan[0]):
+        raise ValueError(
+            "Statistic profile is flat therefore no best-fit value can be determined."
+        )
+
+    result = scipy.optimize.minimize_scalar(
+        interp, bounds=(value_scan[0], value_scan[-1]), method="bounded"
+    )
+    if not result.success:
+        raise RuntimeError("Failed to find minimum in interpolated profile.")
+
+    norm_best_fit = result.x
+    stat_best_fit = result.fun
```

Comment:
Not sure why it fails to find a minimum in the test.
Maybe this would be more stable but it use a different interpolation method so might not be 100% consistent:
```python
    f = InterpolatedUnivariateSpline(value_scan, stat_scan, k=4)
    roots = f.derivative().roots()
    roots = np.append(roots, (value_scan[0], stat_scan[-1]))
    stat _roots= f(roots)
    idx = np.argmin(_roots)
    stat_best_fit = stat[idx]
    norm_best_fit = roots[idx]
```